src/Forms/AssetField.php

```diff
@@ -27,7 +26,8 @@
  *  - Files can't be edited once uploaded.
  *  - Attached files can only be removed, not deleted.
  */
-class AssetField extends FileField {
+class AssetField extends FormField {
+	use Uploadable;
```

Comment:
Having a separate "trait inheritance tree" doesn't really aid readability, but I can't think of an easy alternative (other than rewriting upload behaviour to use composition and a separate class, which would be quite intrusive)